components/data/data.c

```diff
@@ -1,7 +1,20 @@
 #include <stdio.h>
 #include "data.h"
 
-void func(void)
+#define SEPARATOR "|"
+
+bluetooth_frame* new_frame()
 {
+    return malloc(sizeof(bluetooth_frame));
+}
 
+int calc_len(bluetooth_frame* frame)
+{
+    return strlen(frame -> name) + strlen(frame -> mac) + (frame -> rssi == 0 || frame -> rssi == 1 ? 1 : ceil(log10(frame -> rssi))) + 4;
 }
+
+char* format_data(char* str, bluetooth_frame* frame)
```

Comment:
Despite the fact that there hasn't been any documentation in this repo yet, there ought to be. Can you write some docstrings for this? What's `str` supposed to be? It's clear if you look at line 18, but please document it.